tensorflow/python/kernel_tests/eig_op_test.py

```diff
@@ -182,8 +189,70 @@ def Test(self):
   return Test
 
 
+class EigGradTest(test.TestCase):
+  pass  # Filled in below
+
+
+def _GetEigGradTest(dtype_, shape_, compute_v_):
+
+  def Test(self):
+    np.random.seed(1)
+    n = shape_[-1]
+    batch_shape = shape_[:-2]
+    np_dtype = dtype_.as_numpy_dtype
+
+    def RandomInput():
+      # most of matrices are diagonalizable # TODO
```

Comment:
Since the set of non-diagonalizable matrices have measure zero, I think we can leave it as is. It seems to be much easier to return tests in a very unlikely scenario of generating non-diagonalizable than to and complicated checks to the tests.